grafana/datasource.py

```diff
@@ -0,0 +1,52 @@
+import __builtin__
+import json
+import os
+import traceback
+from requests import post
+from grafana.connection import fread
+from grafana.connection import port_open
+from grafana.exceptions import FileNotFoundException
+
+HEADERS = {"Accept": "application/json",
+           "Content-Type": "application/json"
+           }
+
+''' Create Datasource '''
+
+
+def post_datasource(datasource_json):
+    NS = getattr(__builtin__, "NS")
+    if port_open(NS.conf.port, NS.conf.host):
+        resp = post("http://{}:{}/api/datasources"
+                    .format(NS.conf.host,
+                            NS.conf.port),
+                    headers=HEADERS,
+                    auth=NS.conf.credentials,
+                    data=datasource_json)
+
+    return resp
+
+
+def create_datasource(datasource_dir, datasource_name):
+    NS = getattr(__builtin__, "NS")
+    datasource_path = os.path.join(datasource_dir,
+                                   "{}.conf.yml".format(datasource_name))
+    if os.path.exists(datasource_path):
+        datasource_data = fread(datasource_path)
+        try:
+            cfg = NS.datasource_conf
```

Comment:
NS.datasource_conf is set by the python function calling create_datasource() function.